### DISCUSSION TITLE
sleeping objects and sensor contacts

### DISCUSSION THREAD

**Discussion by sherief** [jrouwe/JoltPhysics#280] [Q&A] [answered]
I'm using sensor bodies to implement trigger volumes, and the sensor bodies are created with mAllowSleeping = false and mMotionType = Static. A contact listener's OnContactAdded() and OnContactRemoved() are used to track rigid bodies entering and exiting the trigger volume.

Ideally I'd love to see consistent enter / exit, but I noticed that a rigid body sleeping inside the volume causes a contact removed callback to fire, and so with my implementation I can see objects get enter / exit messages without leaving the volume when they activate / deactivate. 

Is this an issue with my implementation? Is there a better way to get consistent enter / exit notifications for a sensor volume?

**Comment by jrouwe** [comment, marked as answer]
Please see the help text @ [Body::SetIsSensor](https://jrouwe.github.io/JoltPhysics/class_body.html#a5d8e1f66eed163868a633cdb93859f77)
